src/autoscaler/db/sqldb/scheduler_sqldb.go

```diff
@@ -82,3 +83,23 @@ func (sdb *SchedulerSQLDB) GetActiveSchedules() (map[string]*models.ActiveSchedu
 	return schedules, nil
 
 }
+
+func (sdb *SchedulerSQLDB) SynchronizeActiveSchedules(appIdMap map[string]bool) error {
```

Comment:
Since we are keeping this, it should be renamed to what it actually does.
1. This only does Pruning and should follow the conventions of the Pruners.
1. Let us do this the right way and grab the active app ids and compare. We can easily build a loop which will remove X number at a time where X can start with 50. This way we can use the index.